packages/image_picker/ios/Classes/ImagePickerPlugin.m

```diff
@@ -254,4 +319,94 @@ - (UIImage *)scaledImage:(UIImage *)image
   return scaledImage;
 }
 
+- (void)createImageFileAtPath:(NSString *)path contents:(NSData *)data {
+  if ([[NSFileManager defaultManager] createFileAtPath:path contents:data attributes:nil]) {
+    _result(path);
+  } else {
+    _result([FlutterError errorWithCode:@"create_error"
+                                message:@"Temporary file could not be created"
+                                details:nil]);
+  }
+  _result = nil;
+  _arguments = nil;
+}
+
+- (void)createImageFileAtPath:(NSString *)path contents:(NSData *)data with:(PHAsset *)asset {
+  NSMutableDictionary *exifDict = [self fetchExifFrom:asset];
+  NSMutableDictionary *gpsDict = [self fetchGpsFrom:asset];
+
+  NSMutableData *imageData = [NSMutableData data];
+  CGImageSourceRef cgImage = CGImageSourceCreateWithData((__bridge CFDataRef)data, NULL);
+  CGImageDestinationRef destination = CGImageDestinationCreateWithData(
+      (__bridge CFMutableDataRef)imageData, CGImageSourceGetType(cgImage), 1, nil);
+  CGImageDestinationAddImageFromSource(
+      destination, cgImage, 0,
+      (__bridge CFDictionaryRef)[NSDictionary
+          dictionaryWithObjectsAndKeys:exifDict,
+                                       (__bridge NSString *)kCGImagePropertyExifDictionary, gpsDict,
+                                       (__bridge NSString *)kCGImagePropertyGPSDictionary, nil]);
+  CGImageDestinationFinalize(destination);
+
+  [imageData writeToFile:path atomically:YES];
+
+  CFRelease(cgImage);
+  CFRelease(destination);
+
+  _result(path);
+  _result = nil;
+  _arguments = nil;
+}
+
+- (NSMutableDictionary *)fetchExifFrom:(PHAsset *)asset {
```

Comment:
If a manual parser is inevitable, let's put the meta data handling into a separate file/class.